lib/appsignal/config.ex

```diff
@@ -56,10 +56,7 @@ defmodule Appsignal.Config do
   Returns whether the AppSignal agent is configured to start on application launch.
   """
   @spec active?() :: boolean
-  def active? do
-    config = Application.fetch_env!(:appsignal, :config)
-    config.valid && config.active
-  end
+  def active?, do: Application.fetch_env!(:appsignal, :config).active
```

Comment:
This implementation differs from the Ruby gem and I'm not sure we should change that.

https://github.com/appsignal/appsignal-ruby/blob/4ad818d75f113530817ee14af3b3e2af4052b728/lib/appsignal/config.rb#L140-L142

The config can only be active if the config is valid.

Reply:
Exactly. This is the only place `Appsignal.Config.active?/0` is used (which now corresponds to the value in the `:active` configuration and the `APPSIGNAL_ACTIVE` environment variable).

https://github.com/appsignal/appsignal-elixir/blob/7af417d705f7935358ed1f500995ea9ca833a046/lib/appsignal.ex#L65-L89

`Config.initialize()` will return `{:error, :invalid_config}` when the configuration is invalid, disabling the integration. 

I could rename this function, but I need a way to determine if the integration was configured to be active here.

Reply:
Will rename (`should_be_active?/0`) and keep `active?/0` as is.

Reply:
Moving PR back to WIP until the changes are made.